### ISSUE TITLE
508-defect-4: Update birth date to memorable date input fields

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#47460]
### Point of contact

Josh Kim

### Severity level

4, Minor. Consider fixing or exploring in 2-4 sprints post-launch.

### Details

<img width="439" alt="Screen Shot 2022-09-28 at 2 33 05 PM" src="https://user-images.githubusercontent.com/1094951/192861477-d969eb23-e423-4dd3-9c63-328244172d69.png">

[Updating to the memorable date field ](https://design.va.gov/components/form/memorable-date) will make it easier for Veterans to input their birthday as they won't be forced to use a dropdown.

### Reproduction steps

1. [Start the pre need flow as an unsigned in user](https://www.va.gov/burials-and-memorials/pre-need/form-10007-apply-for-eligibility/applicant-information)
2. On the applicant info page, confirm the birth day fields are dropdowns

### Proposed solution or next steps

[Update to the new memorable date field](https://design.va.gov/components/form/memorable-date)

### References, articles, or WCAG support

1. [Memorable date (VA Design System)](https://design.va.gov/components/form/memorable-date)
2. [Memorable date (USWDS)](https://designsystem.digital.gov/components/memorable-date/)

### Type of issue

- [ ] Screenreader
- [ ] Keyboard
- [ ] Focus
- [ ] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by jaredcunha** [comment]
@joshkimux updated your comment by adding a screenshot

**Comment by joshkimux** [comment]
<img width="369" alt="Screenshot 2023-02-17 at 8 49 53 AM" src="https://user-images.githubusercontent.com/14154792/219672189-c2e99acc-e868-419d-a6df-c81cb87dc2a4.png">

@MichelleDieudonne , defect remains and has not been fixed yet. This is a relatively minor defect (level 4), which means its helping improve the usability of the experience (and not a legal requirement). Feel free to deprioritize. The CAIA team (@coforma-terry ) can help provide more support if needed ❇️

**Comment by MichelleDieudonne** [comment]
Jira work item link: https://vajira.max.gov/browse/MBMS-37691

**Comment by ConnorJeff** [comment]
## @joshkimux we have a bug in our pre-need eligibility form when trying to use the VaMemorableDate component:

Staging using the default DateWidget: enter in date 1/1/1000 and click away and validation "Please enter a year between 1900 and 2023' is triggered

https://github.com/department-of-veterans-affairs/va.gov-team/assets/122034971/2fecdd19-1805-4d8b-ac14-b808e8bb795b

Locally using the VaMemorableDate component: enter in date 1/1/1000 and click away, validation doesn't trigger till hitting Continue

https://github.com/department-of-veterans-affairs/va.gov-team/assets/122034971/ef1a15d4-0299-4044-804b-6bb4feae23b0

This is a bug because we are having to use our form's validation system instead of the VaMemorableDate error field. So this has raised the following questions:

- Is using the VaMemorableDate optional? Meaning is it necessary for us to implement this component?
- If so, is this a legitimate bug that would stop us from pushing this through to staging?

**Comment by joshkimux** [comment]
@ConnorJeff @MichelleDieudonne great questions! ✨ 

**Is using the VaMemorableDate optional? Meaning is it necessary for us to implement this component?**
- It is not required, as this is a `defect-4` which is more of a usability consideration than a bug. I recommend attending VA accessibility orientation to learn more about this or reading through platform documentation 🙏 

**If so, is this a legitimate bug that would stop us from pushing this through to staging?**
- Again, this does not block a launch. Platform can only block launches for defect levels 0-1. They strongly recommend fixing defect 2s before launch, but they can be fixed within 1-2 sprints post-launch as well.

**When encountering bugs, please log them with the design system team**
[You can write bug tickets using this template.](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/new?assignees=&labels=vsp-design-system-team%2C+bug&projects=&template=1_bug_template.md&title=) If you search that repo, [you'll notice that this has a number of other bugs as well.](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1593)

**When y'all have time, [platform recommends attending orientations](https://depo-platform-documentation.scrollhelp.site/getting-started/calendly-orientation-sessions)**
They may answer alot of these questions ahead of time!

**also a gentle reminder for the future that I'm technically no longer contractually supporting this**
- I love y'alls team and I really want to help, but I need to work overtime to do so given the volume of product teams across VA.gov and the team I'm specifically assigned to
- questions should be raised in either `#accessibility-help` or 
- through CAIA support tickets 🙇 (where we can prioritize requests and provide a specialist to help y'all when bandwidth is available)  @artsymartha68 and @coforma-terry for awareness 🙇 

**Another pathway for help is through our #a11y-champs program**
To learn more about accessibility and get trained up to learn how to answer these in our growing community, y'all may be interested in joining our accessibility champions program on the #a11y-champs channel! 

I hope this answers y'alls questions, and apologies again I don't have the bandwidth to fully assist 🙇

**Comment by MichelleDieudonne** [comment]
@coforma-terry @artsymartha68 Based on the information that Josh provided above and the work that we put in to try to fix this issue, we have invalidated this bug on our end since this a "usability consideration" and not a true bug.  Can we please get this associated GitHub item closed?  Thank you!

**Comment by joshkimux** [comment]
@MichelleDieudonne and @ConnorJeff 

A couple of updates for y'all, I had more time to look into this with @humancompanion-usds , and have more context to add here 💙 

### [We were unable to replicate this behavior on storybooks](https://design.va.gov/storybook/iframe.html?id=components-va-memorable-date--default&viewMode=story).
The component should be throwing an error on blur after entering an invalid year. We're not sure what is causing the gap between the storybooks instance and y'alls implementation on pre-need.

https://github.com/department-of-veterans-affairs/va.gov-team/assets/14154792/d0b55910-b6bb-4344-a3d1-34bc25aa1394

### Validation on submit is not a bug, and could be the desired state
Validation on submit (as opposed to on blur) which y'all are encountering here may not be consistent with the previous date's component, but it is likely not a bug as the design system team is exploring if this may be a potentially desired state [(e.g. gov uk recommendations)](https://design-system.service.gov.uk/patterns/validation/).

### Whether or not behavior is acceptable is determined through the design system team
Once again, per my previous comment, [y'all can write bug tickets using this template.](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/new?assignees=&labels=vsp-design-system-team%2C+bug&projects=&template=1_bug_template.md&title=) if you do believe it is still a bug, and the design system team can handle it. According to the design system team, this is experimental mostly because teams have yet to adopt it fully (it was added in August), and not because it is defective or incorrect.

**Comment by MichelleDieudonne** [comment]
Memorials Self-Service Jira Work Items: 

- https://jira.devops.va.gov/browse/MBMS-37691
- https://jira.devops.va.gov/browse/MBMS-42695

**Comment by MichelleDieudonne** [comment]
Our research is complete and this work will be completed with our integration work.
